scripts/integration_test.sh

```diff
@@ -14,40 +14,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Setup environment for CI.
 set -e
 
-# Runs integration tests on a given runtime image
+readonly dir=`dirname $0`
 
-dir=`dirname $0`
-projectRoot=$dir/..
-testAppDir=$projectRoot/test-application
-deployDir=$testAppDir/target/deploy
+GetStatusOfContainerBuild () {
+  if [ "$1" -eq 0  ]; then
+    echo "PASSED"
+  else
+    echo "FAILED"
+  fi
+}
 
 imageUnderTest=$1
 if [ -z "${imageUnderTest}" ]; then
   echo "Usage: ${0} <image_under_test>"
   exit 1
 fi
 
-# build the test app
-pushd $testAppDir
-mvn clean install
-popd
-
-# deploy to app engine
-pushd $deployDir
-export STAGING_IMAGE=$imageUnderTest
-envsubst < Dockerfile.in > Dockerfile
-echo "Deploying to App Engine..."
-gcloud app deploy -q
-popd
-
-DEPLOYED_APP_URL="http://$(gcloud app describe | grep defaultHostname | awk '{print $2}')"
-echo "Running integration tests on application that is deployed at $DEPLOYED_APP_URL"
-
-# run in cloud container builder
-gcloud container builds submit \
-  --config $dir/integration_test.yaml \
-  --substitutions "_DEPLOYED_APP_URL=$DEPLOYED_APP_URL" \
-  $dir
+${dir}/ae_integration_test.sh ${imageUnderTest}
+AE_OUTPUT=$?
+AE_STATUS=$(GetStatusOfContainerBuild "$AE_OUTPUT")
+
+${dir}/gke_integration_test.sh ${imageUnderTest}
+GKE_OUTPUT=$?
+GKE_STATUS=$(GetStatusOfContainerBuild "$GKE_OUTPUT")
+
+echo "App Engine integration tests: $AE_STATUS"
+echo "Container Engine integration tests: $GKE_STATUS"
 
```

Comment:
I think that's not necessary, since the script has `set -e` at the top. It will exit immediately if any command exits with non-zero status. 

If that's not what @cassand intends here, then `set -e` should be removed and we can compute the exit code of this script based on how `ae_integration_test` and `gke_integration_test` finish.